test/e2e/checkpoint_test.go

```diff
@@ -1339,4 +1339,35 @@ var _ = Describe("Podman checkpoint", func() {
 		Expect(podmanTest.NumberOfContainersRunning()).To(Equal(0))
 	})
 
+	It("podman checkpoint and restore container with --file-locks", func() {
+		localRunString := getRunString([]string{ALPINE, "flock", "test.lock", "sleep", "100"})
+		session := podmanTest.Podman(localRunString)
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(Exit(0))
+
+		// Checkpoint is expected to fail without --file-locks
+		result := podmanTest.Podman([]string{"container", "checkpoint", "-l"})
+		result.WaitWithDefaultTimeout()
+		Expect(result).Should(Exit(125))
+		Expect(podmanTest.NumberOfContainersRunning()).To(Equal(1))
+
+		// Checkpoint is expected to succeed with --file-locks
+		result = podmanTest.Podman([]string{"container", "checkpoint", "-l", "--file-locks"})
```

Comment:
Does this need a custom crun? It just fails for me with:
```console
$ sudo bin/podman container checkpoint 3a62 --file-locks
checkpoint: unrecognized option '--file-locks'
Try `checkpoint --help' or `checkpoint --usage' for more information.
Error: `/usr/bin/crun checkpoint --image-path /var/lib/containers/storage/overlay-containers/3a62ce7ff8f88b739d192eafedea5717cb224e1bfac8e573ea3ed85dd5ce468e/userdata/checkpoint --work-path /var/lib/containers/storage/overlay-containers/3a62ce7ff8f88b739d192eafedea5717cb224e1bfac8e573ea3ed85dd5ce468e/userdata --file-locks 3a62ce7ff8f88b739d192eafedea5717cb224e1bfac8e573ea3ed85dd5ce468e` failed: exit status 64
```

Reply:
It should work with the patch in https://github.com/containers/crun/pull/783, but it hasn't been released yet.

Reply:
Then there's not much likelihood that CI will pass. (Current failures seem to be a registry outage. I'm not going to hit Re-run since there's no point).

Reply:
I've added a condition to run the test only with `runc` and a FIXME comment to enable the test for `crun`.